src/librustc/ty/constness.rs

```diff
@@ -63,13 +103,81 @@ impl<'tcx> TyCtxt<'tcx> {
 
 
 pub fn provide(providers: &mut Providers<'_>) {
-    /// only checks whether the function has a `const` modifier
+    /// Const evaluability whitelist is here to check evaluability at the
+    /// top level beforehand.
+    fn is_const_intrinsic(tcx: TyCtxt<'_>, def_id: DefId) -> Option<bool> {
+        match tcx.fn_sig(def_id).abi() {
+            Abi::RustIntrinsic |
+            Abi::PlatformIntrinsic => {
+                match tcx.item_name(def_id) {
```

Comment:
Should we de-dup this list with the min-const-fn whitelist by having this one first consult the min-const-fn list?

Reply:
@oli-obk said he'd do some clean-up in a follow-up PR. For now we just want to get our const check in order.

Reply:
Can we leave a FIXME tho?